internal/app/machined/pkg/controllers/cluster/discovery_service.go

```diff
@@ -239,6 +241,9 @@ func (ctrl *DiscoveryServiceController) Run(ctx context.Context, r controller.Ru
 				TTL:           defaultDiscoveryTTL,
 				Insecure:      discoveryConfig.TypedSpec().ServiceEndpointInsecure,
 				ClientVersion: version.Tag,
+				TLSConfig: &tls.Config{
```

Comment:
is this something we missed earlier?

Reply:
yes, I did the update in the client, this will support live reload of TLS certs, but afaik no one waits for it